src/main/java/eu/dissco/backend/domain/jsonapi/JsonApiLinksFull.java

```diff
@@ -18,11 +20,45 @@ public JsonApiLinksFull(int pageNum, int pageSize, boolean hasNext, String path)
     this.next = (hasNext) ? path + pn + (pageNum + 1) + ps + pageSize : null;
   }
 
-  public JsonApiLinksFull(String path){
+  public JsonApiLinksFull(String path) {
     this.self = path;
     this.first = null;
     this.next = null;
     this.prev = null;
   }
 
+  public JsonApiLinksFull(MultiValueMap<String, String> params, int pageNum, int pageSize,
+      boolean hasNext, String path) {
+    params.remove("pageSize");
+    params.remove("pageNumber");
+    var builder = new StringBuilder(path);
+    var isFirst = true;
+    for (var entry : params.entrySet()) {
+      for (var value : entry.getValue()) {
+        if (isFirst) {
+          builder.append("?").append(entry.getKey()).append("=").append(value);
+          isFirst = false;
+        } else {
+          builder.append("&").append(entry.getKey()).append("=").append(value);
+        }
+      }
+    }
+    var pathFull = builder.toString();
+
+    this.self = buildPathPages(pathFull, pageNum, pageSize, isFirst);
+    this.first = buildPathPages(pathFull, 1, pageSize, isFirst);
+    this.prev = (pageNum <= 1) ? null : buildPathPages(pathFull, (pageNum - 1), pageSize, isFirst);
+    this.next = (hasNext) ? buildPathPages(pathFull, (pageNum + 1), pageSize, isFirst) : null;
+  }
+
+  private String buildPathPages(String baseString, int pageNum, int pageSize, boolean isFirst) {
```

Comment:
Nice method here, can you reuse it in the other constructor? (Line 14)

Reply:
I don't really like the MultiValue map why would you make this special map why you could have a Map<X, List<X>> doesn't make sense to me... But Spring requires the MultiValue map so we need that to get it parsed in but would rather look something clearer for the next bit.